app/src/main/java/org/mozilla/focus/session/SessionManager.java

```diff
@@ -260,4 +260,17 @@ public void removeCurrentSession() {
 
         this.sessions.setValue(sessions);
     }
+
+    // Replaces a (custom-tab) session by its no custom-tab version
+    public void replaceSessionWithNoCustomTab(Session session) {
+        if (!session.isCustomTab()) {
+            return;
+        }
+
+        Session noCustomTabSession = new Session(Source.MENU, session.getUrl().getValue());
+        noCustomTabSession.setBlockingEnabled(session.isBlockingEnabled());
+
+        this.addSession(noCustomTabSession);
```

Comment:
As per your comment about crashing, add a comment mentioning this crash and why the ordering is necessary.